### ISSUE TITLE
Parent or Condensed Fee 

### ISSUE THREAD

**Issue by chriscct7** [easydigitaldownloads/Easy-Digital-Downloads#3615]
In order for a fee to apply in a way that it porportionally affects the products it's based on (for example when Discounts Pro has to apply discounts across all affected products) you have to use product based Fees in our Fees API. This lets the discounts be applied in such a way that it simply works for commission calculations as well as accurately reflects the effect of a discount based Fee on all of the affected products. However, the downside is to do this, by applying a fee that is attached to a cart item, each "fee" takes up 1 line item per product. As such, a 3 item cart that has a discount applied to take 20% off the cart total when all three are in the cart requires 3 fees from our Fee API, which takes up 3 rows of fees. This can be confusing to customers who just want to see the overall total of the discount next to the discount name (1 line not 3). We should offer a way in our Fees API that on checkout, in reciepts, and on the purchase confirmation page, we can group discounts together, so they remain behind the scenes separately declared fees, but to customers show as 1 fee.

**Comment by chriscct7** [comment]
Also this could be done at the same time as a proposal (GitHub ticket somewhere) to move Discounts to using the Fees API, at the same time as #2288  and also at the same time as #1848

**Comment by pippinsplugins** [comment]
@chriscct7 Is this what you mean?

Currently:

- Fee 1 - $10
- Fee 2 - $12

Proposal:

- Fees - $22

Then being able to know behind the scenes what the break down is for the total fees?

**Comment by chriscct7** [comment]
Not a total for all fees, the ability to group some of them
Currently:

- Fee 1 for product A - $10
- Fee 1 for product B - $12
- Fee 2 - $10

Proposal:

 - Fee 1 - $22
 - Fee 2 - $10

**Comment by pippinsplugins** [comment]
Ah.

That could be cool, perhaps `parent_fee_id` or something like that.

**Comment by chriscct7** [comment]
Agreed. This would also make it easy to move discounts to using the Fees API, which would then let commission like extensions correctly calculate for per product fees without additional calculations being required. Like for example if we wanted to add a report in the backend that compared fees/discounts to item price

**Comment by SDavisMedia** [comment]
Related ticket: https://secure.helpscout.net/conversation/117924628/16100/?folderId=180499
